Bump multiverse to latest, fix various bugs and improve performance

mobeigi opened the pull request:
# Changes

- Bump package version to `5.0.0`
- Update to latest Multiverse by HTML5 UP (updated on: `07/03/2022`)
  - Available for download at: https://html5up.net/multiverse
  - This bumps jQuery and other js / css files etc
  - Bumps Font Awesome to `v5.15.4`
- Bump jekyll & github-pages in Gemfile
- Remove `webrick` from Gemfile (seems unused?)
- Remove `ads.txt` (seems unused?)
- Reapply Ram's dynamic caption fix to `jquery.poptrox.js v2.5.2-dev`
- Change gulpfile to be a module (.mjs)
- Update gulp to minify all JS & CSS files automatically
- Replace empty brokten email form with generic email mailto template
- Reorder `_config.yml` and use full classname for font-awesome v5 icons
 - This is needed because in v5+ the icons are in different groups based on style
- Defer footer scripts for performance
- Preload font-awesome and fonts in header to make it non-blocking (no CSS `@import`)
- Set `is-preload` in default.html to prevent initial page load flicker (fixes #114 )

# Screenshot
Note the changed email section.

![chrome_7G57rlCLCQ](https://github.com/user-attachments/assets/05128ad0-bc31-4ec3-814c-f913b1481848)


# Review notes
- Can you confirm if `webbrick` in Gemfile is used for anything
- Can you confirm if `ads.txt` was needed for anything
- Note that `jquery.poptrox.js` is the minified variant. I have contacted @ajlkn from (https://github.com/ajlkn/jquery.poptrox) to see if he will share unminified version of `v2.5.2-dev`
- Do you share a public email? I put `email@example.com` but not sure what you prefer for the rampatra version.
- My change to the email section makes the `### Contact Form` section irrelevant. Personally I believe a static email is better for a site like this instead of a form. I'm not sure if you are happy with this or would prefer a revert.
- Unrelated to this PR but the `Presentify` sponsor logo url is broken (because the Github signature probably expired). You can fix that in a different PR.
- Otherwise, please test these changes yourself. I am using these changed on https://photos.mobeigi.com without issues.

CC: @rampatra

rampatra commented:
Okay, I will make some slight changes and approve. But it would be great if you can keep the functionality of the website same as before like the forms, etc. A few folks have said it earlier that it's helpful as one can easily plug that form to a form handler saas without much coding so in those cases it's quite handy. Yea, ads.txt was recommended by Google Ads but we do not need it now as there are no Google Ads shown. Thanks for removing it.

rampatra commented:
Also, I do not understand why/how the spacing between the "Sponsor" section and the "Follow me on..." section is now gone. (from your screenshot attached)

mobeigi commented:
Sure thing we can add the contact form back in. Let me know if you are doing it as part of your changes or I can to.

I'm not sure why the spacing may have broken for that section (maybe the new `.css`). I'll check it later when I have some time to find out why.

mobeigi commented:
Hey Ram,

I made the following changes:

- Restored the email form to what it looked like before
- Fixed the `sponsor` and `follow me on` spacing. It was caused by some missing custom CSS for the footer. I added this to a `custom.scss` file that is compiled with gulp and used rather than changing multiverse files directly (to allow easier future updates).
- Author of poptrox emailed me original `jquery.poptrox.js v2.5.2-dev` so I included the original in source code

mobeigi commented:
Hey @rampatra,

Let me know if you have a chance to review the above changes when you are not busy 👍

add1ct3dd commented:
@rampatra can we consider these additions? These look great and fix up a lot of the older versions :)

rampatra commented on _includes/footer.html:
```diff
@@ -1,9 +1,8 @@
-<script src="assets/js/main.min.js"></script>
\ No newline at end of file
+<script src="assets/js/jquery.min.js" defer></script>
+<script src="assets/js/jquery.poptrox.modified.min.js" defer></script>
```
@mobeigi can you briefly say what is the change in the modified version? can't we go with the version that the author has?

rampatra replied:
This is what Copilot said:

```
GitHub Copilot Chat Assistant — summary of changes in the modified version:

High-level
- Bumped the header/version info (v2.5.1 → v2.5.2-dev) and author/source metadata.
- Modernized and simplified internal code, fixed some jQuery deprecations, and added a visible "loading" class on the popup while content loads.
- Small API/behavior changes that may be breaking (see notes).

Concrete changes (important ones)
- updateWH: converted from function updateWH() to var updateWH = function() { ... } and replaced $(window).width/height with cached $window.width()/$window.height(). Keeps window sizing logic but uses cached jQuery object.
- Loading state:
  - Added $popup.addClass('loading') in multiple places (on open and on switch).
  - Removed some legacy spinner/loader assumptions and ensures loading class is applied while images load.
- Image load handlers:
  - Replaced deprecated .load(fn) with .on('load', fn) and .off('load') for unbinding.
- Captions:
  - Removed runtime invocation of settings.caption (function) — instead the code always uses x.captionText (the captured title). The lazy caption callback path and local variable c were removed.
  - This means custom caption callbacks (if you were using settings.caption as a function) are no longer executed.
- Anchor/title/href handling:
  - i.attr('title', '') became i.removeAttr('title') (removes the title attribute instead of setting it to empty).
  - a.attr('href', '') became a.removeAttr('href') and a CSS cursor: pointer is applied; outline is set to 0. (So anchors no longer have href at all.)
- Exposing settings and return value:
  - $this.prop("_poptrox", settings) changed to $this.prop('_poptrox', settings) with a comment and returns $this (the jQuery object) instead of $(this).
- Minor variable cleanup:
  - Removed unused variable c and an object property "a: a" (the original anchor reference) from x.
  - Consolidated some var declarations (removed "var x, img, c;" → "var x, img;").
- Popup sizing and show/hide:
  - When fixed popup is used, the $popup removes the 'loading' class earlier and sets width/height to popup settings.
  - When dynamic sizing, $popup.removeClass('loading') is called once size is computed and data('width')/data('height') are set.

Possible breaking/behavioral effects to watch for
- Any code relying on settings.caption being a function will no longer get called.
- Removing href from anchors changes semantics (keyboard behavior, right-click menu, etc.) — though cursor:pointer is set to indicate it's clickable.
- CSS reliance on the previous loader/spinner may need updates because a loading class is added instead of relying solely on other loader behavior.
- If external code expected $(element).prop("_poptrox") (with underscore differences only in quotes it's same key but the return type changed to return $this explicitly rather than $(this) — largely equivalent but worth noting).
```

So now my question is, did you do these changes or the original author made these changes and didn't release the update at the time?

rampatra commented:
All other changes look good to me. Once my last comment is answered, I can go ahead and merge this.
